internal/tofu/eval_for_each.go

```diff
@@ -40,7 +40,9 @@ func evaluateForEachExpression(expr hcl.Expression, ctx EvalContext) (forEach ma
 
 // evaluateForEachExpressionValue is like evaluateForEachExpression
 // except that it returns a cty.Value map or set which can be unknown.
-func evaluateForEachExpressionValue(expr hcl.Expression, ctx EvalContext, allowUnknown bool) (cty.Value, tfdiags.Diagnostics) {
+// The 'allowTuple' argument is used to support evaluating for_each from tuple
+// values, and is currently supported when using for_each in import blocks.
+func evaluateForEachExpressionValue(expr hcl.Expression, ctx EvalContext, allowUnknown bool, allowTuple bool) (cty.Value, tfdiags.Diagnostics) {
```

Comment:
minor suggestion: a lot of `evaluateForEachExpressionValue` calls look like this:

```
evaluateForEachExpressionValue(expr, ctx, true, true)
```

It's not obvious what literal boolean values stand for and it could be easier to read with some constants like:

```
evaluateForEachExpressionValue(expr, ctx, unknownsAllowed, tuplesAllowed)
```

Reply:
I agree, though would rather see this change in a followup PR.

Reply:
Totally agree. I'll make sure to add this change to the follow-up PR.